Mention upcoming removal and rename of datasets in docs + release notes (#3155)
kedro-org/kedro: docs/source/data/advanced_data_catalog_usage.md

You can define a Data Catalog in two ways. Most use cases can be through a YAML configuration file as [illustrated previously](./data_catalog.md), but it is possible to access the Data Catalog programmatically through [`kedro.io.DataCatalog`](/kedro.io.DataCatalog) using an API that allows you to configure data sources in code and use the IO module within notebooks.

```{warning}

> Starting from Kedro version `0.19.0`, datasets will no longer be included in the core Kedro package. Instead, you will need to import them from the [`kedro-datasets`](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets) package.

Reviewer: ```suggestion
From Kedro version **`0.19.0`**, datasets are not included in the core Kedro package. Import them from the [`kedro-datasets`](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets) package instead.
```

> Furthermore, beginning with version `2.0.0` of kedro-datasets, all dataset names have been modified to exclude the capital letter "S" in the dataset part of the name. For example, `CSVDataSet` is now `CSVDataset`.

Reviewer: ```suggestion
From version **`2.0.0`** of `kedro-datasets`, all dataset names have changed to replace the capital letter "S" in "DataSet" with a lower case "s". For example, `CSVDataSet` is now `CSVDataset`.
```
